arviz/data/io_pyjags.py

```diff
@@ -0,0 +1,326 @@
+#         _convert_pyjags_samples_dictionary_to_arviz_samples_dictionary(samples)
+#
+#     return az.data.base.dict_to_dataset(data=arviz_dict)
+
+
+# def _convert_pyjags_samples_to_arviz_inference_data_object(
+#         posterior: tp.Optional[tp.Dict[str, np.ndarray]] = None,
+#         prior: tp.Optional[tp.Dict[str, np.ndarray]] = None) \
+#         -> InferenceData:
+#     groups = {}
+#     if posterior is not None:
+#         groups["posterior"] = \
+#             _construct_xarray_dataset_from_pyjags_samples(posterior)
+#     if prior is not None:
+#         groups["prior"] = \
+#             _construct_xarray_dataset_from_pyjags_samples(prior)
+#
+#     return InferenceData(**groups)
+
+
+def _extract_samples_dictionary_from_arviz_inference_data(idata) \
+        -> tp.Dict[str, np.ndarray]:
+    """
+    Extract the samples dictionary from an ArviZ inference data object.
+
+    Extracts a dictionary mapping parameter names to NumPy arrays of samples
+    with shape (number_of_chains, chain_length, parameter_dimension) from an
+    ArviZ inference data object.
+
+    Parameters
+    ----------
+    idata: An Arviz inference data object
+
+    Returns
+    -------
+    a dictionary mapping variable names to NumPy arrays with shape
+             (number_of_chains, chain_length, parameter_dimension)
+
+    """
+    variable_name_to_samples_map = {}
+
+    for key, value in idata.posterior.to_dict()['data_vars'].items():
+        variable_name_to_samples_map[key] = np.array(value['data'])
+
+    return variable_name_to_samples_map
+
+
+def _convert_arviz_samples_dictionary_to_pyjags_samples_dictionary(
+        samples: tp.Dict[str, np.ndarray]) \
+        -> tp.Dict[str, np.ndarray]:
+    """
+    Convert and ArviZ dictionary to a PyJAGS dictionary.
+
+    Takes a python dictionary of samples in ArviZ format and returns the samples
+    as a dictionary in PyJAGS format.
+
+    Parameters
+    ----------
+    samples: a dictionary mapping variable names to NumPy arrays with shape
+             (number_of_chains, chain_length, parameter_dimension)
+
+    Returns
+    -------
+    a dictionary mapping variable names to NumPy arrays with shape
+             (parameter_dimension, chain_length, number_of_chains)
+
+    """
+    # pyjags returns a dictionary of NumPy arrays with shape
+    #         (parameter_dimension, chain_length, number_of_chains)
+    # but arviz expects samples with shape
+    #         (number_of_chains, chain_length, parameter_dimension)
+
+    variable_name_to_samples_map = {}
+
+    for variable_name, chains in samples.items():
+        if chains.ndim == 2:
+            number_of_chains, chain_length = chains.shape
+            chains = chains.reshape((number_of_chains, chain_length, 1))
+
+        variable_name_to_samples_map[variable_name] = \
+            np.swapaxes(chains, 0, 2)
+
+    return variable_name_to_samples_map
+
+
+def from_pyjags(posterior: tp.Optional[tp.Dict[str, np.ndarray]] = None,
```

Comment:
add `dims` and `coords` option too, this way idata can contain metadata for needed dimensions/items.

Extra groups can be added with varnames option (e.g. move certain values from posterior to prediction groups (not sure how pyjags handle these.

Is there a way to calculate elementwise log_likelihood for data?

Also data for example can be added as a dict.

Reply:
I did not find a way to automatically get the log-likelihood from JAGS.